File: idea-plugin/src/main/java/org/ifinalframework/jetbrains/plugins/aio/api/idea/action/OpenMarkdownAction.java
package org.ifinalframework.jetbrains.plugins.aio.api.idea.action;


import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.CommonDataKeys;
import com.intellij.openapi.util.NlsActions;
import com.intellij.psi.PsiElement;
import org.ifinalframework.jetbrains.plugins.aio.$;
import org.ifinalframework.jetbrains.plugins.aio.api.markdown.MarkdownOpenApplication;
import org.ifinalframework.jetbrains.plugins.aio.application.annotation.ReadAction;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * OpenMarkdownAction
 *
 * @author iimik
 * @issue 1
 * @since 0.0.1
 **/
public class OpenMarkdownAction extends AnAction {
    public OpenMarkdownAction() {
    }

    public OpenMarkdownAction(@Nullable Icon icon) {
        super(icon);
    }

    public OpenMarkdownAction(@Nullable @NlsActions.ActionText String text) {
        super(text);
    }

    @Override
    @ReadAction
    public void actionPerformed(@NotNull AnActionEvent anActionEvent) {
        final PsiElement psiElement = anActionEvent.getData(CommonDataKeys.PSI_ELEMENT);
        $.run(MarkdownOpenApplication.class, psiElement);
    }
}

File: idea-plugin/src/main/kotlin/org/ifinalframework/api/idea/action/OpenMarkdownAction.kt
package org.ifinalframework.api.idea.action;

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.CommonDataKeys
import com.intellij.openapi.application.ReadAction
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiElement
import com.itangcent.idea.plugin.actions.BasicAnAction
import com.itangcent.intellij.context.ActionContext
import org.ifinalframework.api.idea.markdown.MarkdownOpener
import javax.swing.Icon


/**
 * OpenMarkdownAction
 *
 * @author iimik
 * @since 2025-01-08 20:42
 **/
class OpenMarkdownAction : BasicAnAction {
    constructor() : super()
    constructor(icon: Icon?) : super(icon)
    constructor(text: String?) : super(text)
    constructor(text: String?, description: String?, icon: Icon?) : super(text, description, icon)
    override fun actionPerformed(actionContext: ActionContext, project: Project?, anActionEvent: AnActionEvent) {
        super.actionPerformed(actionContext, project, anActionEvent)
        actionContext.runInReadUI {
            val element = anActionEvent.getData(CommonDataKeys.PSI_ELEMENT) as PsiElement
            actionContext.instance(MarkdownOpener::class).open(element)
        }
    }

}